src/conn/conn_layered.c

```diff
@@ -1460,6 +1474,221 @@ __layered_drain_ingest_tables(WT_SESSION_IMPL *session)
+ *     Update the timestamp we can prune the ingest tables.
+ */
+static int
+__layered_update_gc_ingest_tables_prune_timestamps(WT_SESSION_IMPL *session)
+{
+    WT_BTREE *ingest;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_DISAGGREGATED_STORAGE *ds;
+    WT_LAYERED_TABLE *layered_table;
+    WT_LAYERED_TABLE_MANAGER *manager;
+    WT_LAYERED_TABLE_MANAGER_ENTRY *entry;
+    size_t i, len, table_count, uri_alloc;
+    wt_timestamp_t prune_timestamp;
+    int64_t ckpt_inuse, last_ckpt, min_ckpt_inuse;
+    uint32_t track;
+    char *uri_at_checkpoint;
+
+    conn = S2C(session);
+    manager = &conn->layered_table_manager;
+    ds = &conn->disaggregated_storage;
+    min_ckpt_inuse = ds->ckpt_track_cnt;
+    uri_at_checkpoint = NULL;
+    uri_alloc = 0;
+
+    __wt_spin_lock(session, &manager->layered_table_lock);
+
+    table_count = manager->open_layered_table_count;
+
+    for (i = 0; i < table_count; i++) {
+        if ((entry = manager->entries[i]) != NULL) {
+            layered_table = entry->layered_table;
+            WT_ERR_NOTFOUND_OK(
+              __layered_last_checkpoint_order(session, layered_table->stable_uri, &last_ckpt),
+              true);
+            /*
+             * If we've never seen a checkpoint, then there's nothing in the ingest table we can
+             * remove. Move on.
+             */
+            if (ret == WT_NOTFOUND)
+                continue;
+
+            /*
+             * For each layered table, we want to see what is the oldest checkpoint on that table
+             * that is in use by any open cursor. Even if there are no open cursors on it, the most
+             * recent checkpoint on the table is always considered in use. The basic plan is to
+             * start with the last checkpoint in use that we knew about, and check it again. If it's
+             * no longer in use, we go to the next one, etc. This gives us a list (possibly zero
+             * length), of checkpoints that are no longer in use by cursors on this table. Thus, the
+             * timestamp associated with the newest such checkpoint can be used for GC pruning. Any
+             * item in the ingest table older than that timestamp must be including in one of the
+             * checkpoints we're saving, and thus can be removed.
+             */
+            ckpt_inuse = layered_table->last_ckpt_inuse;
+            if (ckpt_inuse == 0) {
+                /*
+                 * If we've never checked this layered table before, it's safe to start at the
+                 * oldest checkpoint that we're tracking. There are no cursors in the system that
+                 * are open at checkpoints older than that one. It's probably impossible that we
+                 * haven't tracked any checkpoints, if that happens, we'll start checking at zero.
+                 */
+                if (ds->ckpt_track_cnt > 0)
+                    ckpt_inuse = ds->ckpt_track[ds->ckpt_track_cnt - 1].ckpt_order;
```

Comment:
It was ckpt_inuse = ds->ckpt_track[track].ckpt_order; and it failed compilation. @ddanderson I'm not sure what I did is correct or not.

Reply:
Thanks for flagging this.  The oldest checkpoint is actually always in` ds-ckpt_track[0]`, so I fixed that.